### DISCUSSION TITLE
Path to action in the same repository as the workflow when workflow is inside a different private repository than the triggering workflow

### DISCUSSION THREAD

**Discussion by fbarbare** [community/community#70532] [Actions]
### Select Topic Area

Product Feedback

### Body

Here is the scenario:
Repo A contains a workflow (workflowA) that would be triggered by a GitHub event.
Repo B contains a re-usable workflow (workflowB) and a re-usable action (actionA)

When workflowA gets triggered, it uses workflowB and workflowB uses actionA.

Now there are 2 ways for workflowB to use actionA:
1. use the syntax `uses: octocat/my-private-repo/.github/actions/actionA@main`
2. checkout repo B and point to the folder of action A using a relative path

Using option 1 makes it hard to test any changes because of the `@ref` specification, that means that whenever we want to change something we need to also change the refs and then change them back before merging. Devs forget to change those when testing (meaning they run tests agains main) and it can get quite complex when having multiple re-usable workflows and actions calling each-other.

Using option 2 means that in each job that you run within your workflows you need to first checkout repo B which increases the length and complexity of workflowB. Also, workflowB is already running meaning that repo B was already checked out (in `/home/runner/work/_actions/octocat/my-private-repo/branch-name/`), so why would we need to check it out again?

Could we get a way to point to actionA from workflowB (using relative paths or automatically pointing to the same branch if the `@ref` was omitted or any other implementation) that would automatically point to the branch already in use by workflowB?

**Comment by pcolby** [comment]
I've just run into the same issue. Seems there's more discussion here: https://github.com/orgs/community/discussions/70532
